src/sql/workbench/services/query/common/queryModelService.ts

```diff
@@ -409,6 +416,27 @@ export class QueryModelService implements IQueryModelService {
 		}
 	}
 
+	public async changeConnectionUri(newUri: string, oldUri: string): Promise<void> {
+		// Get existing query runner
+		let queryRunner = this.internalGetQueryRunner(oldUri);
+		if (!queryRunner) {
+			this._logService.error(`A Query and QueryRunner was not found for '${oldUri}'`);
+			throw new Error(nls.localize('queryModelService.noQueryFoundForUri', 'No Query found for {0}', oldUri));
+		}
+		else if (this._queryInfoMap.has(newUri)) {
```

Comment:
And here with overwriting the new connection

Reply:
So basically the execution will try and call QueryModelService's changeConnectionUri, if it fails there, nothing happens and the error handling in queryEditorInput will deal with the changes. If it fails in queryManagement.ts, nothing changes and it goes all the way back to queryEditorInput. At this point, it should be good to run the connectionUriChanged action on the query in STS.